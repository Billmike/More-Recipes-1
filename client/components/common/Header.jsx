import React from 'react';

import Navbar from './Navbar';
import SideNav from './SideNav';

const Header = props => (
  <div>
    <div className="navbar-fixed">
      <nav className="deep-orange darken-4">
        <Navbar {...props} />
      </nav>
    </div>
    <SideNav />
  </div>
);

export default Header;
