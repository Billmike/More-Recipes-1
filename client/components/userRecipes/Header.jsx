import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import SideNav from '../common/SideNav';

/**
 * Class representing Header component
 *
 * @class Header
 * @extends {React.Component}
 */
class Header extends React.Component {
  /**
   * Creates an instance of Header.
   *
   * @memberof Header
   */
  constructor() {
    super();
    this.handleLogoutUser = this.handleLogoutUser.bind(this);
  }
  /**
   * Component did mount lifecycle method
   *
   * @returns {function} Initialize materialize componenets
   * @memberof Header
   */
  componentDidMount() {
    // Initialize materialize css side nav menu activator
    $('.button-collapse').sideNav({
      draggable: true,
    });
    // Initialize materialize dropdown class
    $('.dropdown-button').dropdown({
      belowOrigin: true,
    });
    // Initailize materialize tab class
    $('ul.tabs').tabs();
  }

  /**
   * Logout user from the application
   *
   * @returns {any} Logout user
   * @memberof Header
   */
  handleLogoutUser() {
    const { logoutUser, history } = this.props;
    logoutUser();
    history.push('/');
  }

  /**
   * Render method
   *
   * @returns {object} React element
   * @memberof Header
   */
  render() {
    return (
      <div>
        <div className="navbar-fixed">
          <nav className="nav-extended deep-orange darken-4">
            <div className="nav-wrapper" >
              <Link
                to="/"
                id="logo"
                className="brand-logo hide-on-small-only"
              >
                More-Recipes
              </Link>
              <a
                data-activates="slide_out"
                className="button-collapse"
              >
                <i className="material-icons">menu</i>
              </a>
            </div>
            <div className="nav-content">
              <ul className="tabs tabs-transparent">
                <li className="tab"><a href="#user-profile">Profile</a></li>
                <li className="tab"><a href="#user-recipes">My Recipes</a></li>
                <li className="tab">
                  <a href="#user-favorites">My Favorites</a>
                </li>
              </ul>
            </div>
          </nav>
        </div>
        <SideNav {...this.props} />
      </div>
    );
  }
}

Header.propTypes = {
  logoutUser: PropTypes.func.isRequired,
  history: PropTypes.shape({
    push: PropTypes.func.isRequired,
  }).isRequired,
};

export default Header;
