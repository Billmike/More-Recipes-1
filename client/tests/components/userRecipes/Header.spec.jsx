import React from 'react';

import Header from '../../../components/userRecipes/Header';

describe('Header component', () => {
  let spy;

  afterEach(() => {
    spy.mockClear();
  });
  it('should render component', () => {
    spy = jest.spyOn(Header.prototype, 'componentDidMount');
    const props = {
      logoutUser: jest.fn(),
      history: {
        push: jest.fn(),
      },
      handleLogoutUser: jest.fn(),
      user: {
        isAuthenticated: true,
      },
    };
    const wrapper = shallow(<Header {...props} />);

    expect(wrapper).toBeDefined();
    expect(spy).toHaveBeenCalled();
    expect(wrapper).toMatchSnapshot();
  });
});
