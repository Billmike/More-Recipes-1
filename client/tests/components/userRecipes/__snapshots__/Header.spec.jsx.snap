// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render component 1`] = `
<div>
  <div
    className="navbar-fixed"
  >
    <nav
      className="nav-extended deep-orange darken-4"
    >
      <div
        className="nav-wrapper"
      >
        <Link
          className="brand-logo hide-on-small-only"
          id="logo"
          replace={false}
          to="/"
        >
          More-Recipes
        </Link>
        <a
          className="button-collapse"
          data-activates="slide_out"
        >
          <i
            className="material-icons"
          >
            menu
          </i>
        </a>
      </div>
      <div
        className="nav-content"
      >
        <ul
          className="tabs tabs-transparent"
        >
          <li
            className="tab"
          >
            <a
              data-tab="user-profile"
              href="#user-profile"
            >
              Profile
            </a>
          </li>
          <li
            className="tab"
          >
            <a
              data-tab="user-recipes"
              href="#user-recipes"
            >
              Recipes
            </a>
          </li>
          <li
            className="tab"
          >
            <a
              data-tab="user-favorites"
              href="#user-favorites"
            >
              Favorites
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </div>
  <SideNav
    authenticatedUserId={null}
    currentProfileUserId={null}
    handleLogoutUser={[MockFunction]}
    history={
      Object {
        "push": [MockFunction],
      }
    }
    logoutUser={[MockFunction]}
    user={
      Object {
        "isAuthenticated": true,
      }
    }
  />
</div>
`;
